src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -4290,6 +4290,90 @@ def _check_zip_deployment_status(cmd, rg_name, name, deployment_status_url, slot
     return res_dict
 
 
+def _get_latest_deployment_id(cmd, rg_name, name, deployment_status_url, slot):
+    import requests
+    from azure.cli.core.util import should_disable_connection_verify
+
+    headers = get_scm_site_headers(cmd.cli_ctx, name, rg_name, slot)
+    total_trials = 30
+    num_trials = 0
+    while num_trials < total_trials:
+        time.sleep(2)
+        response = requests.get(deployment_status_url, headers=headers,
+                                verify=not should_disable_connection_verify())
+        try:
+            res_dict = response.json()
+        except Exception as ex:  # pylint: disable=broad-except
+            logger.warning("Deployment status endpoint %s returned malformed data. Exception: %s "
+                           "\nRetrying...", deployment_status_url, ex)
+            return None
+        finally:
+            num_trials = num_trials + 1
+        if 'id' in res_dict and 'temp' not in res_dict['id']:
+            return res_dict['id']
+    return None
+
+
+def _track_deployment_runtime_status(params, deploymentstatusapi_url, timeout=None):
+    total_trials = (int(timeout) // 5) if timeout else 200
```

Comment:
Did We check if timeout is working as expected?

Reply:
yeah, tested it and it works fine